VikingQuiz.Api/Controllers/QuizzesController.cs

```diff
@@ -70,121 +71,172 @@ public class QuizzesController : Controller
             }
 
             string fileUrl;
             AzureBlobService blobService;
-            try {
-                blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
+            try
+            {
+                blobService = new AzureBlobService(azureContainerName);
+                fileUrl = await blobService.UploadPhotoAsync(quizBodyData.Files[0]);
             }
-            catch (Exception) {
+            catch (Exception)
+            {
                 return BadRequest("Image could not be uploaded.");
             }
 
 
-            Quiz createdQuiz = quizRepository.CreateQuiz(new Quiz {
+            Quiz createdQuiz = quizRepository.CreateQuiz(new Quiz
+            {
                 Title = quizBodyData.Title,
                 PictureUrl = fileUrl,
-                UserId = currentUserId
+                UserId = currentUserId,
+                LastModified = DateTime.Now
             });
-            if (createdQuiz == null) {
+            if (createdQuiz == null)
+            {
                 return BadRequest("Quiz couldn't be created");
             }
 
             QuizViewModel quizVm = entityToVmMapper.Map(createdQuiz);
-            quizVm.PictureUrl = blobService.urlPath.AbsoluteUri.ToString() + "users/" + fileUrl;
+            quizVm.PictureUrl = new AzureBlobService(azureContainerName).GetFullUrlOfFileName(quizVm.PictureUrl);
 
-            return Ok(new { quizVm.Id, quizVm.Title, quizVm.PictureUrl });
+            return Ok(quizVm);
         }
 
 
         // update a specific quiz id
         [HttpPost("{id}")]
         [RequestSizeLimit(5_000_000)]
         [Authorize]
-        public async Task<IActionResult> Put(NewQuizViewModel quizBodyData, int id) {
+        public async Task<IActionResult> UpdateQuiz(NewQuizViewModel quizBodyData, int id)
+        {
+
+            IActionResult response = Ok();
             int currentUserId = User.Claims.GetUserId();
+
             IActionResult imageHttpResponse = FileValidityChecker(quizBodyData);
-            if (IActionResult.Equals(imageHttpResponse, Ok())) {
+            if (OkResult.Equals(imageHttpResponse, response) )
+            {
                 return imageHttpResponse;
             }
 
-            string fileUrl;
-            AzureBlobService blobService;
-            try {
-                blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                var previousQuizState = quizRepository.GetQuizById(id);
-
-                blobService.DeletePhoto(previousQuizState.PictureUrl);
-                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
-            }
-            catch (Exception) {
-                return BadRequest("Image could not be uploaded.");
+            UploadAndDeleteData uploadAndDeleteResult = await uploadAndDeletePhoto(id, quizBodyData.Files[0]);
```

Comment:
uploadAndDeletePhoto
Uploads and delete uplaoded photo or correct is "OverritePhoto" or "AddOrUpdate" or "DeleteAndUpload" <- to actions is bad but sometimes is acceptble